src/types/redis_cms.cc

```diff
@@ -0,0 +1,264 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include "redis_cms.h"
+
+#include <stdint.h>
+
+#include "cms.h"
+#include "rocksdb/status.h"
+
+namespace redis {
+
+rocksdb::Status CMS::GetMetadata(engine::Context &ctx, const Slice &ns_key, CountMinSketchMetadata *metadata) {
+  return Database::GetMetadata(ctx, {kRedisCountMinSketch}, ns_key, metadata);
+}
+
+rocksdb::Status CMS::IncrBy(engine::Context &ctx, const Slice &user_key,
+                            const std::unordered_map<std::string, uint64_t> &elements) {
+  std::string ns_key = AppendNamespacePrefix(user_key);
+
+  LockGuard guard(storage_->GetLockManager(), ns_key);
+  CountMinSketchMetadata metadata{};
+  rocksdb::Status s = GetMetadata(ctx, ns_key, &metadata);
+
+  if (s.IsNotFound()) {
+    return rocksdb::Status::NotFound();
+  }
+  if (!s.ok() && !s.IsNotFound()) {
+    return s;
+  }
+
+  auto batch = storage_->GetWriteBatchBase();
+  WriteBatchLogData log_data(kRedisCountMinSketch);
+  batch->PutLogData(log_data.Encode());
+
+  CMSketch cms(metadata.width, metadata.depth, metadata.counter, metadata.array);
+
+  if (elements.empty()) {
+    return rocksdb::Status::OK();
+  }
```

Comment:
I don't understand why it would be empty: https://github.com/RedisBloom/RedisBloom/blob/b5e674f61e84f9636b26d2005fe5ea50230ecb79/src/rm_cms.c#L120